### PR TITLE
8253053: Javadoc clean up in Authenticator and BasicAuthenicator

### PR DESCRIPTION
**Pull request by pconcannon**
Hi,

Could someone please review my doc-only fix for JDK-8253053 - 'Javadoc clean up in Authenticator and BasicAuthenicator' ?

This fix is set of formatting changes intended to clean up the javadoc of the following classes :

`com.sun.net.httpserver.Authenticator`
`com.sun.net.httpserver.Authenticator.Result`
`com.sun.net.httpserver.Authenticator.Failure`
`com.sun.net.httpserver.Authenticator.Success`
`com.sun.net.httpserver.Authenticator.Retry`

`com.sun.net.httpserver.BasicAuthenticator` 

This issue is a sub-task of [JDK-8252822](https://bugs.openjdk.java.net/browse/JDK-8252822)

Kind regards,
Patrick

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253053](https://bugs.openjdk.java.net/browse/JDK-8253053): Javadoc clean up in Authenticator and BasicAuthenicator

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**) ⚠️ Review applies to 5d9004612a60da62af3b30502ef2a1d8dfce2826
 * [Michael McMahon](https://openjdk.java.net/census#michaelm) (@Michael-Mc-Mahon - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/301/head:pull/301`
`$ git checkout pull/301`

### CODE REVIEW

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java [hunk 1]
```diff
@@ -74,7 +72,7 @@ public Failure (int responseCode) {
         /**
          * returns the response code to send to the client
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java:73]: I'd prefer to make this a full sentence: 
`* Returns the response code to send to the client.`

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java [hunk 2]
```diff
@@ -92,15 +90,15 @@ public int getResponseCode() {
...
          * Creates a {@code Success} instance with given {@code Principal}.
          *
-         * @param p The authenticated user you wish to set as Principal
+         * @param p the authenticated user you wish to set as {@code Principal}
          */
         public Success (HttpPrincipal p) {
             principal = p;
         }
         /**
          * returns the authenticated user Principal
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java:99]: `* Returns the authenticated user {@code Principal}.`

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java [hunk 3]
```diff
@@ -132,7 +130,7 @@ public Retry (int responseCode) {
         /**
          * returns the response code to send to the client
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java:131]: `* Returns the response code to send to the client.`

**Review by dfuch — CHANGES_REQUESTED** [review]
The first sentence in an API documentation comment should be a full sentence.

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java (hunk 3 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java]: Change made in https://github.com/openjdk/jdk/pull/301/commits/604ea45ca36c8fb0cb195c581f822dc5ae1c23c1

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java (hunk 2 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java]: Change made in https://github.com/openjdk/jdk/pull/301/commits/604ea45ca36c8fb0cb195c581f822dc5ae1c23c1

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java (hunk 1 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java]: Change made in https://github.com/openjdk/jdk/pull/301/commits/604ea45ca36c8fb0cb195c581f822dc5ae1c23c1

**Review by RogerRiggs — APPROVED** [review]
Looks fine.  Thanks.

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/BasicAuthenticator.java [hunk 4]
```diff
@@ -130,14 +131,14 @@ private void setAuthHeader(HttpExchange t) {
     }
 
     /**
-     * called for each incoming request to verify the
+     * Called for each incoming request to verify the
      * given name and password in the context of this
      * Authenticator's realm. Any caching of credentials
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/BasicAuthenticator.java:136]: Maybe remove the uppercase in "this Authenticator's realm"

**Review by dfuch — COMMENTED** [review]
Looks generally good - just one minor comment below...

**Review by ChrisHegarty — APPROVED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/BasicAuthenticator.java (hunk 4 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/BasicAuthenticator.java]: Change made in https://github.com/openjdk/jdk/pull/301/commits/5d9004612a60da62af3b30502ef2a1d8dfce2826

**Review by dfuch — APPROVED** [review]
Looks good Patrick!

**Review by RogerRiggs — APPROVED** [review]
fyi, the trailing </li> can be omitted.  It is not required and the source is more readable without it.

**Review by ChrisHegarty — APPROVED** [review]

**Comment by pconcannon** [comment]
> fyi, the trailing can be omitted. It is not required and the source is more readable without it.

Thanks for the tip, Roger! I've made that change and you can view it in the latest commit (https://github.com/openjdk/jdk/pull/301/commits/b6328074ebc7cb1099f0788940714fd80501bed7)

**Review by dfuch — APPROVED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java [hunk 5]
```diff
@@ -122,41 +121,40 @@ public HttpPrincipal getPrincipal() {
...
-     * <p>
-     * Sucess means that the authentication
-     * has succeeded, and a Principal object representing the user
-     * can be retrieved by calling Sucess.getPrincipal() .
-     * <p>
-     * Retry means that another HTTP exchange is required. Any response
-     * headers needing to be sent back to the client are set in the
-     * given HttpExchange. The response code to be returned must be provided
-     * in the Retry object. Retry may occur multiple times.
+     * Called to authenticate each incoming request. The implementation
```
Comment (Michael-Mc-Mahon) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java:142]: Should there be closing < /l i> tags here?

Comment (Michael-Mc-Mahon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Authenticator.java:142]: Sorry, I see you had them in already, but removed them. Never mind.

**Review by Michael-Mc-Mahon — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]
Looks good, thanks

**Pull request closed without merging**